uvloop/__init__.py

```diff
@@ -34,69 +37,84 @@ def install() -> None:
     __asyncio.set_event_loop_policy(EventLoopPolicy())
 
 
-def run(main, *, loop_factory=new_event_loop, debug=None, **run_kwargs):
-    """The preferred way of running a coroutine with uvloop."""
-
-    async def wrapper():
-        # If `loop_factory` is provided we want it to return
-        # either uvloop.Loop or a subtype of it, assuming the user
-        # is using `uvloop.run()` intentionally.
-        loop = __asyncio._get_running_loop()
-        if not isinstance(loop, Loop):
-            raise TypeError('uvloop.run() uses a non-uvloop event loop')
-        return await main
-
-    vi = _sys.version_info[:2]
-
-    if vi <= (3, 10):
-        # Copied from python/cpython
-
-        if __asyncio._get_running_loop() is not None:
-            raise RuntimeError(
-                "asyncio.run() cannot be called from a running event loop")
-
-        if not __asyncio.iscoroutine(main):
-            raise ValueError("a coroutine was expected, got {!r}".format(main))
-
-        loop = loop_factory()
-        try:
-            __asyncio.set_event_loop(loop)
-            if debug is not None:
-                loop.set_debug(debug)
-            return loop.run_until_complete(wrapper())
-        finally:
+if _typing.TYPE_CHECKING:
```

Comment:
Can we just inline the typing and omit this TYPE_CHECKING check?

Reply:
I tried that, if you see the history of the PR, but it took too many changes

Reply:
I see now. What about adding `new_event_loop` to `__init__.pyi`?

Reply:
I tried that first also and it means the other members aren't type checked